### ISSUE TITLE
Design Parachain/Parathread incentivisation

### ISSUE THREAD

**Issue by eskimor** [paritytech/polkadot-sdk#709]
For parathread core orders, we want some minimal price with the argument that there is some value always and also there is some effort for the network no matter the market conditions. Hence it would make sense for at least the minimum order price to go to validators directly, instead of burning everything.

Ideally from my point of view that fixed amount should go to validators directly - not even to nominators, but directly on top of the commission. This is not strictly required, as at least in theory validators can customize their commission rate freely and thus adjust it to cover their operation costs anyways. (There are issues in practice, so adding order fees to commission directly might actually be very worthwhile.)

But what ever we think for parathreads has to also consider parachains. Parachains right now only pay opportunity costs by not being able to nominate with the locked up DOT in their lease. This superficially works, because validators and nominators earn DOT via inflation - by removing the locked lease token from nominations, there are essentially more rewards for the remaining nominators. So this is basically how parachains pay the validators for their work.

For parathreads, if we just burn order fees, a similar argument holds: Those token cannot be used for nominating anymore, hence a bigger share of the inflation goes to remaining nominators. But there is also another effect: Those DOT are completely and permanently removed from the market, leading to a deflationary effect, which benefits everybody, not just the validators who actually do the work.

How is the situation for parachains on the other hand? Maybe not that different: If we assume a parachain is here to stay and keeps renewing its slot and for simplicity, assuming every time for the same amount of DOT, then those DOT are effectively also removed permanently from the market leading to that deflationary effect*). 

In that vein, it has been discussed: If we scale up more: More validators, more parachains .. then the amount of work for validators will increase, while compensation would not necessarily.

Also related, there is some fixed overhead on the network per parachain block, no matter whether that block does something or is completely empty: This means either that incentives should be aligned so that parathreads are way cheaper on average, if you are not going to have data for almost every block on your parachain - or alternatively we could also charge parachains per block (which should be trivial to implement), incentivizing chains to not produce blocks if they are not filled much or empty. 

The second option is tempting, but also might lead to the network being more prone to fluctuations in availability/performance. We might end up oversubscribing the network, causing issues if multiple chains start producing more blocks at the same time. A similar concern exists now already with load per block, but the issue would definitely be amplified. Could be mitigated by making block price dynamic, but then we would more and more run into similar problems as smart contracts competing for resources on Ethereum!

This problem does not exist for the first variant (at least not for parachains): If there are lots of parathreads wanting to author a block at the same time, they are competing for their set of cores - hence the parathread block price would go up, but parachains would stay unaffected - which is exactly what we want and is one of the trade-offs between parathreads and parachains.

TL;DR, we should figure out:

- [ ] How to fairly distribute order fees?
- [ ] How to align this with parachains, in a way that people are incentivized to use the most applicable tool for the job? 
- [ ] How do we ensure validators stay profitable when scaling up?

Just some thoughts on my end, I am not too much involved into economic considerations, so please correct me if I got some aspect wrong.

*) Things are more nuanced, DOTs staked are also locked and removed from the market, but arguably easier to bring back than DOTs locked in a lively parachain.
